packages/core/src/shared/crashMonitoring.ts

```diff
@@ -324,6 +359,84 @@ class CrashChecker {
 
     /** Mimics a crash, only for testing */
     public testCrash() {
+        this.cleanup()
+    }
+}
+
+/**
+ * This class is required for the following edge case:
+ * 1. Heartbeat is sent
+ * 2. Computer goes to sleep for X minutes
+ * 3. Wake up computer. But before a new heartbeat can be sent, a crash checker (can be from another ext instance) runs
+ *   and sees a stale heartbeat. It assumes a crash.
+ *
+ * Why? Intervals do not run while the computer is asleep, so the latest heartbeat has a "lag" since it wasn't able to send
+ *      a new heartbeat.
+ *      Then on wake, there is a racecondition for the next heartbeat to be sent before the next crash check. If the crash checker
+ *      runs first it will incorrectly conclude a crash.
+ *
+ * Solution: Keep track of the lag, and then skip the next crash check if there was a lag. This will give time for the
+ *           next heartbeat to be sent.
+ */
+class TimeLag {
+    private intervalRef: NodeJS.Timer | undefined
+    private isCompleted: Promise<void> | undefined
+    /** Resolves {@link isCompleted} when the next interval of lag checking is completed */
+    private setCompleted: (() => void) | undefined
+
+    /** The last timestamp we remember. If this is more than 1 {@link lagCheckInterval} we probably did a sleep+wake */
+    private latestTimestamp: number = 0
```

Comment:
these 3 properities are common to the existing Timeout class in our timoutUtils module. does using Timeout save any code?